controllers/protectedvolumereplicationgrouplist_controller.go

```diff
@@ -50,31 +51,36 @@ type ProtectedVolumeReplicationGroupListInstance struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.9.2/pkg/reconcile
 func (r *ProtectedVolumeReplicationGroupListReconciler) Reconcile(ctx context.Context, req ctrl.Request,
 ) (ctrl.Result, error) {
-	log := ctrl.Log.WithName("controllers").WithName("protectedvolumereplicationgrouplist").
-		WithValues("name", req.NamespacedName.Name)
-
-	log.Info("protectedvolumereplicationgrouplist reconcile start")
-
+	start := time.Now()
 	// save all the commonly used parameters in a struct
 	s := ProtectedVolumeReplicationGroupListInstance{
 		reconciler: r,
 		ctx:        ctx,
-		log:        log,
+		log:        ctrl.Log.WithName("pvrgl").WithValues("name", req.NamespacedName.Name),
 		instance:   &ramendrv1alpha1.ProtectedVolumeReplicationGroupList{},
 	}
 
 	// get ProtectedVolumeReplicationGroupListInstance and save to s.instance
-	if err := r.Client.Get(ctx, req.NamespacedName, s.instance); err != nil {
+	if err := r.Client.Get(s.ctx, req.NamespacedName, s.instance); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(fmt.Errorf("get: %w", err))
 	}
 
+	s.log = s.log.WithValues("rid", s.instance.ObjectMeta.UID, "rv", s.instance.ObjectMeta.ResourceVersion)
+	s.ctx = ctrl.LoggerInto(ctx, s.log)
+
+	s.log.Info("reconcile start")
+
+	defer func() {
+		s.log.Info("reconcile end", "time spent", time.Since(start))
```

Comment:
(future) While time spent is useful in the debugging context here, we should really be looking at metrics in the future for the same.

Reply:
ack